app/services/requirements/image_upload_checker.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require "mini_magick"
+
+module Requirements
+  class ImageUploadChecker
+    include ActionView::Helpers::NumberHelper
+
+    SUPPORTED_FORMATS = %w(image/jpeg image/png image/gif).freeze
+    MAX_FILE_SIZE = 20.megabytes
+
+    attr_reader :file
+
+    def initialize(file)
+      @file = file
+    end
+
+    def issues
+      issues = []
+
+      unless file
+        issues << Issue.new(:image_upload, :no_file)
+        return CheckerIssues.new(issues)
+      end
+
+      if unsupported_type? || animated_image?
+        issues << Issue.new(:image_upload, :format_not_allowed)
```

Comment:
One of the things I'd suggest we do (but not a priority) is to split up type and animated into separate checks. I'd suggest the static gif terminology of the message is a bit confusing and it'd be easier to just have a separate message that was "You must... Upload an animate that is not animated"

Reply:
Yeah, I was wondering about that - wanted it to be a pure refactor, though ;-). I've made a note and might get on to it later.

Reply:
Yup, no problem - I missed that first time round so thought I'd drop it in.